transport.go

```diff
@@ -104,6 +104,7 @@ func (t *Transport) handshake(
 	select {
 	case remotePubKey = <-keyCh:
 	default:
+		return nil, errors.New("go-libp2p-tls BUG: expected remote pub key to be set")
```

Comment:
This previously implicitly checked if the channel was closed before returning a key. We still need to check that.

Reply:
Thanks for catching this. Using a chan to return the key seems a bit error-prone...